### ISSUE TITLE
Explore using numba and cython

### ISSUE THREAD

**Issue by hlasimpk** [rigdenlab/SIMBAD#116]
This should improve speed of code when iterating through lists

**Comment by fsimkovic** [comment]
numba is not really an option unless CCP4 started supporting it. not sure where you'd need cython except for reading files though cctbx is about  as good as it gets whilst keeping the Python interface

**Comment by fsimkovic** [comment]
I think you could gain some performance by 
- caching results. any calculation that is based on some input that might be identical to others could be stored in a cache (see `functools.lru_cache`). matthews/solvent/lattice scores/etc.

Apart from that you might struggle to keep the ability to distribute calculations both locally and across a cluster

**Comment by hlasimpk** [comment]
Fair point about numba. I thought I might get some performance improvements when looping through results, but this was just something I wanted to explore when I get a little more time rather than a hard plan

**Comment by fsimkovic** [comment]
had a stab in #117 to improve performance of the lattice search. I noticed that `functools.lru_cache` is not available until py3+ so you're stuck until CCP4 finally decide to upgrade their version.
